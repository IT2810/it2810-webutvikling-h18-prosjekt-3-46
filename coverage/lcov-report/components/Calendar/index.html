<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Calendar</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/Calendar
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.19% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/64</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">19.23% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">24.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/62</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="17.19" class="pic low"><div class="chart"><div class="cover-fill" style="width: 17%;"></div><div class="cover-empty" style="width:83%;"></div></div></td>
	<td data-value="17.19" class="pct low">17.19%</td>
	<td data-value="64" class="abs low">11/64</td>
	<td data-value="19.23" class="pct low">19.23%</td>
	<td data-value="26" class="abs low">5/26</td>
	<td data-value="24.14" class="pct low">24.14%</td>
	<td data-value="29" class="abs low">7/29</td>
	<td data-value="17.74" class="pct low">17.74%</td>
	<td data-value="62" class="abs low">11/62</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 19 2018 20:24:02 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
