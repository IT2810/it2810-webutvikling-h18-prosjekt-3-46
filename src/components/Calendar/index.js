import React, { Component } from 'react';
import {
    Text,
    View,
    AsyncStorage, TextInput, Platform
} from 'react-native';
import {Agenda} from 'react-native-calendars';
import { Button, Icon} from 'native-base';
import Modal from "react-native-modal";

export default class AgendaScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            items: {},
            selectedDate: this.timeToString(new Date()),
            visibleModal: false,
            currentText: "",
        };
    }

    static navigationOptions = ({ navigation }) => {
        const params = navigation.state.params || {};

        return {
            title: 'Calendar',
            headerRight: (
                <Button transparent onPress={params.toggleModal}>
                    <Icon name="add-circle" type={"Ionicons"}/>
                </Button>
            )
        };
    };

    componentWillMount() {
        this.props.navigation.setParams({toggleModal: this._toggleModal});
    }

    render() {
        return (
            <View style={styles.container}>
                <View>
                    <Modal isVisible={this.state.visibleModal} style={styles.modalContent}>
                        <View style={styles.smallContainer}>

                            <Text style={{marginTop:20}}>Edit a note for this day: {this.state.selectedDate}</Text>
                            <View style={styles.textContainer}>
                                <TextInput
                                    style={styles.textInput}
                                    keyboardType = 'default'
                                    returnKeyType="done"
                                    returnKeyLabel="done"
                                    value={ this.itemData(this.state.items[this.state.selectedDate]) }
                                    onChangeText={(text) => this.setState({currentText: text})}
                                />
                            </View>
                            <Button block success onPress={this.addText} style={{marginTop: 20}}>
                                <Text style={{color: "white"}}>Add Note</Text>
                            </Button>
                            <Button block onPress={this._toggleModal} style={{marginTop: 20}}>
                                <Text style={{color: "white"}}>Close</Text>
                            </Button>
                        </View>
                    </Modal>
                </View>

                <Agenda
                    items={this.state.items}
                    loadItemsForMonth={this.loadItems.bind(this)}
                    renderItem={this.renderItem.bind(this)}
                    renderEmptyDate={this.renderEmptyDate.bind(this)}
                    rowHasChanged={this.rowHasChanged.bind(this)}
                    onDayPress={(day) => {this.selectDate(day)}}
                    onDayChange={(day) => {this.selectDate(day)}}
                />
            </View>
        );
    }

    loadItems(day) {
        setTimeout(() => {
            for (let i = -15; i < 50; i++) {
                const time = day.timestamp + i * 24 * 60 * 60 * 1000;
                const strTime = this.timeToString(time);
                if (!this.state.items[strTime]) {
                    this.state.items[strTime] = [];
                    this._retrieveData(strTime);

                }
            }
            const newItems = {};
            Object.keys(this.state.items).forEach(key => {newItems[key] = this.state.items[key];});
            this.setState({
                items: newItems
            });
        }, 1000);
    }

    loadItem(stringDate) {
        setTimeout(() => {
                if (!this.state.items[stringDate]) {
                    this.state.items[stringDate] = [];
                    this._retrieveData(stringDate);
                }
            const newItems = {};
            Object.keys(this.state.items).forEach(key => {newItems[key] = this.state.items[key];});
            this.setState({
                items: newItems
            });
        }, 1000);
    }

    renderItem(item) {
        return (
            <View style={[styles.item, {height: item.height}]}><Text>{item.name}</Text></View>
        );
    }

    renderEmptyDate() {
        return (
            <View style={styles.emptyDate}><Text>Ingen avtaler</Text></View>
        );
    }

    rowHasChanged(r1, r2) {
        return r1.name !== r2.name;
    }

    timeToString(time) {
        const date = new Date(time);
        return date.toISOString().split('T')[0];
    }

    itemData(item) {
        if(item === undefined) {
            return "Ingen avtaler";
        }
        if(item.length === 0) {
            return "Ingen avtaler";
        }
        return item[0].name;
    }

    selectDate(day) {
        this.setState({selectedDate: day.dateString,
                        currentText: this.itemData(this.state.items[this.state.selectedDate])
        });
    }

    _toggleModal = () => {
        let temp = this.state.visibleModal;
        this.setState({visibleModal: !temp});
    };

    addText = () => {
        let temp = this.state.visibleModal;
        this.setState({visibleModal: !temp});
        this._saveData(this.state.selectedDate, this.state.currentText);
        this._retrieveData(this.state.selectedDate);
        this.loadItem(this.state.selectedDate);
    }

    _retrieveData = async (dateKey) => {
        try {
            const note = await AsyncStorage.getItem(dateKey);
            if (note !== null) {
                this.state.items[dateKey] = [];
                this.state.items[dateKey].push({
                    name: note,
                    height: Math.max(50, Math.floor(Math.random() * 150))
                });
            }
        } catch (error) {
        }
    }

    _saveData = (dateKey, note) => {
        AsyncStorage.removeItem(dateKey);
        AsyncStorage.setItem(dateKey, note);
    };
}

const styles = StyleSheet.create({
    item: {
        backgroundColor: 'white',
        flex: 1,
        borderRadius: 5,
        padding: 10,
        marginRight: 10,
        marginTop: 17
    },
    emptyDate: {
        height: 15,
        flex:1,
        paddingTop: 30
    },
    container: {
        flex: 1,
        width: "100%"

    },
    smallContainer: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "#fff",
        width: "100%"
    },
    textContainer: {
        backgroundColor: "white",
        padding: 10,
        width: "100%"
    },
    modalContent: {
        backgroundColor: "white",
        padding: 22,
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 4,
        borderColor: "rgba(0, 0, 0, 0.1)",
        width: "90%"
    },
    textInput: {
        height: 50,
        paddingRight: 15,
        paddingLeft: 15,
        borderColor: "#F2F2F2",
        borderWidth: 2,
        borderRadius: 15,
        backgroundColor: "white",
        fontSize: 18
    },
});