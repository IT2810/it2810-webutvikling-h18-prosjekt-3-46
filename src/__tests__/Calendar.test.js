import React from 'react';

import Calendar from '../screens/Calendar/index';
import renderer from 'react-test-renderer';
import moment from 'moment';


test('renders correctly', ()=> {
   const tree = renderer.create(<Calendar/>).toJSON();
   expect(tree).toMatchSnapshot();
});
