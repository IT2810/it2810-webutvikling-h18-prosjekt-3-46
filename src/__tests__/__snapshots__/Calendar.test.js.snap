// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "rgba(0, 0, 0, 0.1)",
            "borderRadius": 4,
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "padding": 22,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
            "width": "90%",
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          accessible={false}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "letterSpacing": 3,
              }
            }
          >
            19.10.2018
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginTop": 30,
              }
            }
          >
            Edit a note for this day:
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              keyboardType="default"
              multiline={true}
              numberOfLines={5}
              onChangeText={[Function]}
              placeholder="What are your plans?"
              returnKeyLabel="done"
              returnKeyType="done"
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#F2F2F2",
                  "borderRadius": 15,
                  "borderWidth": 2,
                  "fontSize": 18,
                  "height": 150,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#5cb85c",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginTop": 20,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Add Note
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#d9534f",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginTop": 20,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Remove Note
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginTop": 20,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Close
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#f4f4f4",
          "flex": 1,
          "marginTop": 104,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "marginTop": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 1230,
          "height": 0,
          "justifyContent": "flex-end",
          "overflow": "hidden",
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 1230,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "bottom": 0,
            "flex": 1,
            "height": 24,
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#f2F4f5",
                "borderRadius": 3,
                "height": 7,
                "marginTop": 10,
                "width": 38,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 104,
          "justifyContent": "space-around",
          "left": 0,
          "marginLeft": 15,
          "marginRight": 15,
          "opacity": 0,
          "paddingBottom": 7,
          "paddingTop": 15,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": -104,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Sun
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Mon
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Tue
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Wed
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Thu
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Fri
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#b6c1cd",
            "fontSize": 13,
            "textAlign": "center",
            "width": 32,
          }
        }
      >
        Sat
      </Text>
    </View>
  </View>
</View>
`;
